terraform/interpolate.go

```diff
@@ -129,15 +129,6 @@ func (i *Interpolater) valueModuleVar(
 	n string,
 	v *config.ModuleVariable,
 	result map[string]ast.Variable) error {
-	// If we're computing all dynamic fields, then module vars count
-	// and we mark it as computed.
-	if i.Operation == walkValidate {
-		result[n] = ast.Variable{
-			Value: config.UnknownVariableValue,
-			Type:  ast.TypeString,
-		}
-		return nil
-	}
```

Comment:
This behavior was introduced in PR #432.
The latter test still exists, and the first test was removed via PR #2479.